client/lib/plans/personal-plan.js

```diff
@@ -32,18 +44,33 @@ export const personalPlan = {
 	available: 'yes',
 	bundle_product_ids: [ 12, 9, 50, 5, 6, 46, 54, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 72, 73, 74, 75, 16 ],
 	bill_period_label: translate( 'per year' ),
-	price: '$71.88',
-	formatted_price: '$71.88',
-	raw_price: 71.88,
 	has_domain_credit: true
 };
 
+const getCurrencyCode = plans => head( map( plans, property( 'currency_code' ) ) ) || 'USD';
+
+const applyCurrency = currencyCode => {
+	const cost = personalPlan.prices[ currencyCode ];
+	const price = formatCurrency( cost, currencyCode );
+
+	return {
+		...personalPlan,
+		cost,
+		price,
+		raw_price: cost,
+		formatted_price: price,
+		currency_code: currencyCode
+	};
+};
+
+const formatPlan = flow( getCurrencyCode, applyCurrency );
+
 export const insertPersonalPlan = plans => {
 	const freePlanIndex = findIndex( plans, matchesProperty( 'product_slug', PLAN_FREE ) );
-	const hasPersonalPlan = plans.some( matchesProperty( 'product_slug', PLAN_PERSONAL ) );
+	const hasPersonalPlan = some( plans, matchesProperty( 'product_slug', PLAN_PERSONAL ) );
 
 	return ! hasPersonalPlan
-		? [ ...plans.slice( 0, freePlanIndex + 1 ), personalPlan, ...plans.slice( freePlanIndex + 1 ) ]
+		? [ ...plans.slice( 0, freePlanIndex + 1 ), formatPlan( plans ), ...plans.slice( freePlanIndex + 1 ) ]
```

Comment:
Related to the above, `formatPlan( plans )` seems weird because we give it an array and it spits out one particular element in return. Maybe we should grab the personal plan from this list before sending it to `format`

Remember that we can almost always eliminate taking arrays as arguments to our functions.

```js
const currencyCode = ( { currency_code } ) => currency_code || 'USD';
const convertPrices = ( plan, code ) => {
	const cost = plan.prices[ code ];
	const price = formatCurrency( cost, code );

	return {
		...plan,
		cost,
		price,
		…
	}
}

export const insertPersonalPlan = plans => {
	…
	const personalPlan = …

	? [ ...plans.slice( … ), formatPlan( personalPlan ), …plans.slice( … ) ]
}
```